### PR TITLE
WIP: feat: support executing bin again when bin exit

### LINKED ISSUE #353: Air should support executing binary again when executable exit
Currently it doesn't execute binary again when exit. It does only when file changes. `Air` should exit with the same status code as exited binary **OR** rerun binary again with optional configurable delay. Both features should be configurable and may be disabled on default to keep backward compatibility.

**Use case:**
Microservices or long-running applications can be run as services under Docker Compose with the `restart: always` option, Docker Swarm or other container orchestration solution. For development purpose `Air` can be also run in these environments. But some applications/frameworks terminate running application for example because of unreachable message broker or database service instance. And orchestration rerun them again. But with `Air` it never happens. Because `Air` never exit or rerun binary when binary exit by itself.

### PR DESCRIPTION
**Pull request by amikai**
close #353

### CODE REVIEW

**Comment by xiantang** [comment]
work in progress?

File: runner/engine.go [hunk 1]
```diff
@@ -431,31 +431,37 @@ func (e *Engine) building() error {
 }
```
Comment (xiantang) [inline runner/engine.go:432]: we need to add a new flag to support this, and disable this behavior by default

Comment (amikai) [reply runner/engine.go:432]: ok

**Comment by xiantang** [comment]
I mean you need add a test case like this: when executing bin exit should restart the bin again

**Comment by xiantang** [comment]
Do we need to append a document for this?

File: runner/engine_test.go [hunk 2]
```diff
@@ -110,6 +111,87 @@ func TestRegexes(t *testing.T) {
...
+	}
+
+	time.Sleep(time.Second * 1)
+	// stop engine
+	engine.Stop()
+	time.Sleep(time.Second * 1)
+	t.Logf("engine stopped")
+
+	roundAfterChange := atomic.LoadUint64(&engine.round)
+	fmt.Println(roundBeforeChange)
```
Comment (amikai) [inline runner/engine_test.go:188]: delete this

**Comment by amikai** [comment]
> Do we need to append a document for this?

Finish 1c7171469e52d6b693e72794258c36983ea7892e

**Pull request merged**